docs/design/Cephfs-RDR-ConsistencyGroup.md

```diff
@@ -0,0 +1,411 @@
+  meaning all PVCs are not part of any consistency group.
+
+### Does enabling the feature change any default behavior
+
+No, only PVCs included in a volume group will adhere to this design;
+other PVCs will continue to follow the current approach.
+
+### Dependencies
+
+* CephFS VolumeGroupSnapshot CSI
+
+### New API
+
+To support Volume Group during Regional DR,
+two additional CRDs have been introduced:
+
+* RGS (ReplicationGroupSource): This resource functions similarly
+  to volsync ReplicationSource, managing Volume Groups as a source.
+
+```
+type ReplicationGroupSource struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec   ReplicationGroupSourceSpec   `json:"spec,omitempty"`
+  Status ReplicationGroupSourceStatus `json:"status,omitempty"`
+}
+
+type ReplicationGroupSourceSpec struct {
+  // Trigger to do the schedule sync or final sync
+  Trigger *ReplicationSourceTriggerSpec `json:"trigger,omitempty"`
+
+  VolumeGroupSnapshotClassName *string `json:"volumeGroupSnapshotClassName,omitempty"`
+
+  // RSSpecs specify the PVCs in a volume group
+  RSSpecs []VolSyncReplicationSourceSpec `json:"rsspecs,omitempty"`
+}
+
+// ReplicationGroupSourceStatus defines the observed state of ReplicationGroupSource
+type ReplicationGroupSourceStatus struct {
+  // lastSyncTime is the time of the most recent successful synchronization.
+  //+optional
+  LastSyncTime *metav1.Time `json:"lastSyncTime,omitempty"`
+  // lastSyncStartTime is the time the most recent synchronization started.
+  //+optional
+  LastSyncStartTime *metav1.Time `json:"lastSyncStartTime,omitempty"`
+  // lastSyncDuration is the amount of time required to send the most recent
+  // update.
+  //+optional
+  LastSyncDuration *metav1.Duration `json:"lastSyncDuration,omitempty"`
+  // nextSyncTime is the time when the next volume synchronization is
+  // scheduled to start (for schedule-based synchronization).
+  //+optional
+  NextSyncTime *metav1.Time `json:"nextSyncTime,omitempty"`
+  // lastManualSync is set to the last spec.trigger.manual when the manual sync is done.
+  //+optional
+  LastManualSync string `json:"lastManualSync,omitempty"`
+  // conditions represent the latest available observations of the
+  // source's state.
+  Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+* RGD (ReplicationGroupDestination): This resource functions similarly
+  to volsync ReplicationDestination, managing Volume Groups as a destination.
+
+```
+type ReplicationGroupDestination struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec   ReplicationGroupDestinationSpec `json:"spec,omitempty"`
+  Status *ReplicationDestinationStatus   `json:"status,omitempty"`
+}
+
+type ReplicationGroupDestinationSpec struct {
+  // RDSpecs specify the PVCs in a volume group
+  RDSpecs []VolSyncReplicationDestinationSpec `json:"rdspecs,omitempty"`
```

Comment:
This should contain ReplicationDestinations

Reply:
same with ReplicationSources